File: Decorator/Java/PizzaMenu/Barbeque.java
package Decorator.PizzaMenu;

public class Barbeque extends ToppingDecorator {

    Pizza pizza;

    public Barbeque(Pizza pizza) {
        this.pizza = pizza;
    }

    @Override
    public String getName() {
        return "Barbeque " + pizza.getName();
    }

    @Override
    public int getCost() {
        return 50 + pizza.getCost();
    }
}

File: Decorator/Cpp/PizzaMenu/src/Barbeque.cpp
/*
 * Barbeque.cpp
 *
 *  Created on: Jan 18, 2021
 *      Author: nquang
 */

#include "Barbeque.h"

Barbeque::Barbeque(Pizza *component) {
	this->component_ = component;
}
int Barbeque::getCost() const {
	return this->component_->getCost() + this->cost;
}
string Barbeque::getName() const {
	return this->name + this->component_->getName();
}
string Barbeque::getDescription() const {
	return this->component_->getDescription() + this->name;
}
